src/Sarif.Viewer.VisualStudio/ResultTextMarker.cs

```diff
@@ -57,478 +59,293 @@ public ResultTextMarker(int runId, Region region, string fullFilePath)
             }
 
-            Span spanToColor;
-            int markerStart, markerEnd = 0;
 
-            try
-            {
-                // Fix up the end line number if it's inconsistent
-                if (endLine <= 0 || endLine < startLine)
-                {
-                    endLine = startLine;
-                }
-
-                bool coerced = false;
+            _tag.CaretEnteredTag += CaretEnteredTag;
+            _wpfTextView = wpfTextView;
+            _vsWindowFrame = vsWindowFrame;
 
-                // Calculate the start and end marker bound. Adjust for the column values if
-                // the values don't make sense. Make sure we handle the case of empty file correctly
-                ITextSnapshotLine startTextLine = textSnapshot.GetLineFromLineNumber(Math.Max(startLine - 1, 0));
-                ITextSnapshotLine endTextLine = textSnapshot.GetLineFromLineNumber(Math.Max(endLine - 1, 0));
-                if (startColumn <= 0 || startColumn >= startTextLine.Length)
-                {
-                    startColumn = 1;
-                    coerced = true;
-                }
+            return true;
+        }
 
-                // Calculate the end marker bound. Perform coersion on the values if they aren't consistent
-                if (endColumn <= 0 && endColumn >= endTextLine.Length)
-                {
-                    endColumn = endTextLine.Length;
-                    coerced = true;
-                }
+        // When the VS Editor tag has the caret moved inside of it, let's just pass along the region selection.
+        private void CaretEnteredTag(object sender, EventArgs e) => this.RaiseRegionSelected?.Invoke(this, new EventArgs());
 
-                // If we are highlighting just one line and the column values don't make
-                // sense or we corrected one or more of them, then simply mark the
-                // entire line
-                if (endLine == startLine && (coerced || startColumn >= endColumn))
-                {
-                    startColumn = 1;
-                    endColumn = endTextLine.Length;
-                }
+        private static bool TryCreateTextSpanWithinDocumentFromSourceRegion(Region region, IVsWindowFrame vsWindowFrame, out TextSpan textSpan)
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
 
-                // Create a span with the calculated markers
-                markerStart = startTextLine.Start.Position + startColumn - 1;
-                markerEnd = endTextLine.Start.Position + endColumn - 1;
-                spanToColor = Span.FromBounds(markerStart, markerEnd);
+            // SARIF regions are 1 based, VS is one based.)
+            textSpan.iStartLine = Math.Max(region.StartLine - 1, 0);
+            textSpan.iEndLine = Math.Max(region.EndLine - 1, 0);
+            textSpan.iStartIndex = Math.Max(region.StartColumn - 1, 0);
+            textSpan.iEndIndex = Math.Max(region.EndColumn - 1, 0);
```

Comment:
Man this looks like a perfect opportunity for an extension method to set a textSpan from a region.

Reply:
If this were done in any other place, I might consider it.

---
In reply to: [481396731](https://github.com/microsoft/sarif-visualstudio-extension/pull/191#discussion_r481396731) [](ancestors = 481396731)